vue-loader with support of extract styles (#340)

morr opened the pull request:
this PR fixes two things

1. #340 issue
2. Styles parsing in `.vue` should work exactly the same way it works for regular `.css`/`.sass` files. This means that in `loaders/vue` we should use the same loaders configuration that we use in `loaders/sass`. But I'm not sure where to put shared loaders configuration. In this PR I just copy-pasted them.

I also added ability to disable styles extraction with `extractStyles` variable because in some cases it's better to inline styles rather than extract them. 

Any suggestions? How about such file?
`loaders/core/common(or shared or configuration).js`
```js
const cssLoader = [
  { loader: 'css-loader', options: { minimize: env === 'production' } },
  'postcss-loader'
]
module.exports = {
  css: cssLoader,
  sass: cssLoader.concat(['sass-loader?indentedSyntax'])
  scss: cssLoader.concat(['sass-loader'])
}
```

RavenXce commented on lib/install/config/loaders/installers/vue.js:
```diff
@@ -1,13 +1,36 @@
+      use: loader
+    })
+  }
+  return ['vue-style-loader'].concat(sassLoader)
```
Shouldn't this be  `loader` not `sassLoader`?

morr replied:
oh my... yes this should be `loader`

RavenXce commented on lib/install/config/loaders/installers/vue.js:
```diff
@@ -1,13 +1,36 @@
+const sassLoader = cssLoader.concat(['sass-loader?indentedSyntax'])
+const scssLoader = cssLoader.concat(['sass-loader'])
+
+function vueLoader(loader) {
```
Seems like a rather confusing name as this is prepending the `vue-style-loader`, not `vue-loader`.

gauravtiwari commented:
@morr Sorry for late review - please could you add `postcss` too in the loaders list.

morr commented:
@gauravtiwari it was already added to loaders list. You mean this, right?
![monosnap 2017-05-26 14-26-44](https://cloud.githubusercontent.com/assets/134610/26492928/661e8696-421f-11e7-95ab-6b033c8d25b7.png)

I rebased pull request on current `rails/webpacker#master` branch and added to `vue.js` latest loaders changes from `loaders/core/sass.js`

gauravtiwari commented:
@morr Ahh yes! Looks good to me - I will merge 👍  🍰 

Thank you :)

inopinatus commented on lib/install/config/loaders/installers/vue.js:
```diff
@@ -1,13 +1,41 @@
+const { env } = require('../configuration.js')
+
+// Change it to false if you prefer Vue styles to be inlined by javascript in runtime
+const extractStyles = false
```
Does this need `const extractStyles = true` to avoid a regression on #310?
